File: move_zeroes/move_zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        write_pointer = 0
        for val in nums:
            if val != 0:
                nums[write_pointer] = val
                write_pointer += 1
        
        for i in range(write_pointer, len(nums)):
            nums[i] = 0
        

File: move_zeroes/move_zeroes.java
class Solution {
    /* The idea here is really to keep the write pointer method, it will do its thing and loop till end     of the array (first pass).
    At the end of that operation, you have a pointer that tells you:
    - Anything from this pointer -> end is 0.
    So we loop again from the pointer -> end and fill with 0s.
    */
    
    public void moveZeroes(int[] nums) {
        int writePointer = 0;
        for (int i = 0 ; i < nums.length; i++) {
            if (nums[i] != 0) {
                nums[writePointer] = nums[i];
                writePointer++;
                    
            } 
        }
        
        for (int i = writePointer; i < nums.length; i++) {
            nums[i] = 0;
        }
    }
}